src/shims/env.rs

```diff
@@ -130,18 +130,21 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         let size = this.read_scalar(size_op)?.to_usize(&*this.tcx)?;
         // If we cannot get the current directory, we return null
         // FIXME: Technically we have to set the `errno` global too
-        if let Ok(cwd) = env::current_dir() {
-            // It is not clear what happens with non-utf8 paths here
-            let mut bytes = cwd.display().to_string().into_bytes();
-            // If the buffer is smaller or equal than the path, we return null.
-            // FIXME: Technically we have to set the `errno` global too
-            if (bytes.len() as u64) < size {
-                // We add a `/0` terminator
-                bytes.push(0);
-                // This is ok because the buffer is larger than the path with the null terminator.
-                this.memory_mut().get_mut(buf.alloc_id)?.write_bytes(tcx, buf, &bytes)?;
-                return Ok(Scalar::Ptr(buf))
+        match env::current_dir() {
+            Ok(cwd) =>{
+                // It is not clear what happens with non-utf8 paths here
+                let mut bytes = cwd.display().to_string().into_bytes();
+                // If the buffer is smaller or equal than the path, we return null.
+                if (bytes.len() as u64) < size {
+                    // We add a `/0` terminator
+                    bytes.push(0);
+                    // This is ok because the buffer is larger than the path with the null terminator.
+                    this.memory_mut().get_mut(buf.alloc_id)?.write_bytes(tcx, buf, &bytes)?;
+                    return Ok(Scalar::Ptr(buf))
+                }
+                this.machine.last_error = 34; // ERANGE
```

Comment:
uh... is this like a guarantee? I think it would be better to use the same constant fetching scheme as https://github.com/rust-lang/miri/blob/7095a3bb68d0462df28a9ac0b56db155bfd3afcf/src/shims/foreign_items.rs

Reply:
What do you mean?

Reply:
oops that link only went to the file. I meant

https://github.com/rust-lang/miri/blob/7095a3bb68d0462df28a9ac0b56db155bfd3afcf/src/shims/foreign_items.rs#L291

so basically

```
let erange = this.eval_path_scalar(&["libc", "ERANGE"])?;
```

and then getting the integer out of it

Reply:
Ohhh ok